cpp/lib/src/LocalDataDB.cc

```diff
@@ -0,0 +1,215 @@
+    if (result_set.empty())
+        return {}; // empty vector
+
+    const auto row(result_set.getNextRow());
+    return BlobToLocalFieldsVector(row["local_fields"], title_ppn);
+}
+
+
+bool LocalDataDB::removeTitleDataSet(const std::string &title_ppn) {
+    // 1. Clear out any local PPNs associated with "title_ppn":
+    db_connection_->queryOrDie("SELECT local_fields FROM local_data WHERE title_ppn = "
+                               + db_connection_->escapeAndQuoteString(title_ppn));
+    auto result_set(db_connection_->getLastResultSet());
+    if (result_set.empty())
+        return false;
+
+    const auto row(result_set.getNextRow());
+    const auto previous_local_fields(BlobToLocalFieldsVector(row["local_fields"], title_ppn));
+    const auto local_ppns(ExtractLocalPPNsFromLocalFieldsVector(previous_local_fields));
+    for (const auto &local_ppn : local_ppns)
+        db_connection_->queryOrDie("DELETE * FROM local_ppns_to_title_ppns_map WHERE local_ppn="
+                                   + db_connection_->escapeAndQuoteString(local_ppn));
+
+    // 2. Delete the local data for the title PPN:
+    db_connection_->queryOrDie("DELETE FROM local_data WHERE title_ppn = "
+                               + db_connection_->escapeAndQuoteString(title_ppn));
+
+    return true;
+}
+
+
+// Removes all fields from "local_fields" that are associated w/ "local_ppn".
+static std::vector<std::string> RemoveLocalDataSet(const std::string &local_ppn,
+                                                   const std::vector<std::string> &local_fields)
+{
+    std::vector<std::string> filtered_local_fields;
+    filtered_local_fields.reserve(local_fields.size());
+
+    bool skipping(false);
+    for (const auto &local_field : local_fields) {
+        if (StringUtil::StartsWith(local_field, "001 "))
+            skipping = local_field.substr(__builtin_strlen("001 ")) == local_ppn;
+        if (not skipping)
+            filtered_local_fields.emplace_back(local_field);
+    }
+
+    return filtered_local_fields;
+}
+
+
+bool LocalDataDB::removeLocalDataSet(const std::string &local_ppn) {
+    // 1. Determine the title PPN associated w/ the local PPN:
+    db_connection_->queryOrDie("SELECT title_ppn FROM local_ppns_to_title_ppns_map WHERE local_ppn="
+                               + db_connection_->escapeAndQuoteString(local_ppn));
+    auto result_set(db_connection_->getLastResultSet());
+    if (result_set.empty())
+        return false;
+    const auto row(result_set.getNextRow());
+    const auto title_ppn(row["title_ppn"]);
+
+    // 2. Retrieve the local data associated w/ the title PPN:
+    auto local_fields(getLocalFields(title_ppn));
+
+    // 3. Remove the local data associsted w/ the local PPN:
+    const auto filtered_local_fields(RemoveLocalDataSet(local_ppn, local_fields));
+
+    // 4. Update our SQL tables:
+    db_connection_->queryOrDie("DELETE FROM local_ppns_to_title_ppns_map WHERE ocal_ppn="
```

Comment:
typo "ocal_ppn"

Reply:
Fixed in the commit from the ruschein2 branch.